woocommerce/woocommerce-ios
# Order Fulfillment: Allow the user to enter package tracking details using a custom shipment provider

Fix #846 

This PR implements adding shipment tracking using a custom tracking provider. 

![Simulator Screen Shot - iPhone 8 - 2019-04-18 at 14 31 35](https://user-images.githubusercontent.com/2722505/56341158-c101be00-61e6-11e9-879a-37eeb825e878.png)
![Simulator Screen Shot - iPhone 8 - 2019-04-18 at 14 31 52](https://user-images.githubusercontent.com/2722505/56341167-c52ddb80-61e6-11e9-93e4-f1329617197d.png)

## Changes
- Networking: Added a parameter to the relevant method in `ShipmentsRemote` to pass the `shipmentDate`
- Yosemite: Added a parameter to the relevant methods in `ShipmentAction` and `ShipmentStore` to pass the `shipmentDate` when creating a shipment tracking with a custom provider
- `ManualTrackingViewModel`: Add an implementation for this use case, providing its own collection of sections to be rendered by the view controller. Also add and remove some properties to the interface.
- `ManualTrackingViewController`: Add configuration for cells corresponding to the new fields necessary to render the new sections, and add logic to dispatch the action to create a shipment tracking with a custom provider
- Unit tests and updates to the existing tests.

## Testing
- Checkout the branch, run the tests.
- Navigate to an order > Fulfill Order > Add Tracking
- Tap Shipping Provider (first row)
- Filter the list to reach the "error" view with a "Custom Provider" button. Tap the button
- Attempt to create a tracking with any name / tracking number.
- Name and tracking number are mandatory. URL is optional. If a url is entered, the system should  prepend `https://` is it's missing.
- It should still be possible to create tracking using one of the pre-defined providers
- In the Order Details screen, only trackings with url should show a "Track Package" button.
